pkg/build/reaper/reaper.go

```diff
@@ -0,0 +1,114 @@
+package reaper
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/kubectl"
+	kutilerrors "k8s.io/kubernetes/pkg/util/errors"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/util"
+)
+
+// NewBuildConfigReaper returns a new reaper for buildConfigs
+func NewBuildConfigReaper(oc *client.Client) kubectl.Reaper {
+	return &BuildConfigReaper{oc: oc, pollInterval: kubectl.Interval, timeout: kubectl.Timeout}
+}
+
+// BuildConfigReaper implements the Reaper interface for buildConfigs
+type BuildConfigReaper struct {
+	oc                    client.Interface
+	pollInterval, timeout time.Duration
+}
+
+// Stop deletes the build configuration and all of the associated builds.
+func (reaper *BuildConfigReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *kapi.DeleteOptions) (string, error) {
+	noBcFound := false
+	noBuildFound := true
+
+	// Add deletion pending annotation to the build config
+	err := unversioned.RetryOnConflict(unversioned.DefaultRetry, func() error {
+		bc, err := reaper.oc.BuildConfigs(namespace).Get(name)
+		if kerrors.IsNotFound(err) {
+			noBcFound = true
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+
+		// Ignore if the annotation already exists
+		if strings.ToLower(bc.Annotations[buildapi.BuildConfigPausedAnnotation]) == "true" {
+			return nil
+		}
+
+		// Set the annotation and update
+		if err := util.AddObjectAnnotations(bc, map[string]string{buildapi.BuildConfigPausedAnnotation: "true"}); err != nil {
+			return err
+		}
+		_, err = reaper.oc.BuildConfigs(namespace).Update(bc)
+		return err
+	})
+	if err != nil {
+		return "", err
+	}
+
+	// Collect builds related to the config.
+	builds, err := reaper.oc.Builds(namespace).List(buildutil.BuildConfigSelector(name), nil)
+	if err != nil {
+		return "", err
+	}
+	errList := []error{}
+	for _, build := range builds.Items {
+		noBuildFound = false
+		if err := reaper.oc.Builds(namespace).Delete(build.Name); err != nil {
+			glog.Warningf("Cannot delete Build %s/%s: %v", build.Namespace, build.Name, err)
+			if !kerrors.IsNotFound(err) {
+				errList = append(errList, err)
+			}
+		}
+	}
+
+	// Collect deprecated builds related to the config.
+	// Note: Delete this block after BuildConfigLabelDeprecated is removed.
+	builds, err = reaper.oc.Builds(namespace).List(buildutil.BuildConfigSelectorDeprecated(name), nil)
+	if err != nil {
+		return "", err
+	}
```

Comment:
if you error out here, you're not going to print the warning that the BuildConfig isn't being deleted.

I think you need to audit all the return calls in this function to make sure the warning is always getting printed.